lib/commons/builder/wikidata_queries.rb

```diff
@@ -1,243 +1,97 @@
 # frozen_string_literal: true
 
-class WikidataQueries < Wikidata
-  def date_condition(start_date, end_date)
-    return '' unless start_date
-    end_date ||= '9999-12-31'
-    <<~DATE_CONDITION
-      BIND(COALESCE(?start, "1000-01-01T00:00:00Z"^^xsd:dateTime) AS ?start_or_sentinel)
-      BIND(COALESCE(?end, "9999-12-31T00:00:00Z"^^xsd:dateTime) AS ?end_or_sentinel)
-      FILTER (?end_or_sentinel >= "#{start_date}"^^xsd:dateTime)
-      FILTER (?start_or_sentinel <= "#{end_date}"^^xsd:dateTime)
-  DATE_CONDITION
-  end
-
-  def term_condition(term_item_id)
-    return '' unless term_item_id
-    "?statement pq:P2937 wd:#{term_item_id} ."
-  end
-
-  # Use this to generate a `SERVICE wikibase:label` SPARQL pattern, for use in
-  # situations where you want a single label to help the humans that maintain
-  # a democratic commons repository. Do not use it for labels to be presented
-  # to end-users, as that needs potentially multiple language-tagged labels.
-  # For that, use `lang_select` and `lang_options`.
-  def label_service
-    languages_with_en = (['en'] + languages).uniq
-    "SERVICE wikibase:label { bd:serviceParam wikibase:language \"#{languages_with_en.join(',')}\". }"
-  end
+require 'liquid'
 
+class WikidataQueries < Wikidata
   def query_legislative(position_item_id:, house_item_id:, term_item_id: nil, start_date: nil, end_date: nil, **_rest)
-    unless !!term_item_id ^ !!(start_date and end_date)
+    unless term_item_id ^ (start_date and end_date)
```

Comment:
I think the `!!` (or something else) is still needed here to coerce these values to booleans. e.g. if `term_item_id` is present it'll be a string and the XOR will throw an exception:
```
[10] pry(main)> 'Q42' ^ nil
NoMethodError: undefined method `^' for "Q42":String
from (pry):10:in `__pry__'
```

Though the double-bangs are a bit unclear, though, and I think RuboCop doesn't like them. An alternative suggestion adapted from StackOverflow:

```
[15] pry(main)> ['a', nil].one?(&:nil?)
=> true
[16] pry(main)> ['a', 'b'].one?(&:nil?)
=> false
[17] pry(main)> [nil, 'b'].one?(&:nil?)
=> true
[18] pry(main)> [nil, nil].one?(&:nil?)
=> false
```

Reply:
(If that removing of the `!!` didn't cause a test failure, I think it should have!)

Reply:
I don't think that method is tested anywhere :/ (one for the backlog). And yeah, I'll reinstate the doppelbangen.